autode/values.py

```diff
@@ -619,9 +619,8 @@ def __eq__(self, other):
             other is not None
             and hasattr(other, "shape")
             and other.shape == self.shape
-            and np.allclose(self, other)
+            and repr(self) == repr(other)
```

Comment:
this is somewhat of a breaking change, but is probably wrong currently so fine to patch?

Reply:
@t-young31 why not do `np.allclose(np.asarray(self), np.asarray(other))`?

Reply:
because `np.allclose` on ValueArrays  would silently ignore the (kw)arg(s) (e.g. `np.allclose(value_array_1, value_array_1, rtol=0.1)` as `np.allclose` now includes a equality check) 😞

Reply:
@t-young31 But if we set the `np.allclose` rtol and atol to a very low value ( 1E-12) in the ValueArray class then it is not a problem? Because then `np.allclose(value_arr1, value_arr2, rtol=0.1)` would give the expected result because the equality check will return False, but the actual value check will return True.

Reply:
yeah, could do. I guess repr is the same to whatever the default numpy precision is. I don't love a call to `allclose` being duplicated and being wrong if the rtol specified is smaller than is set in the equality :confused. maybe equality should be on the pointers? i.e. `id(self) == id(other)` – what do you think?

Reply:
@t-young31 With `repr` it is still essentially comparing arrays to a set number of decimal points. So default numpy repr prints upto 8 decimal points from what I can see, so is equivalent to calling a`allclose(atol=1e-8)`. So it would still be wrong if someone wants allclose with < 1e-8 tolerance. I guess you could use pointers, that would be consistent with the python default object equality test.

Reply:
@t-young31 If I understand correctly, valuearray equal is almost never used in the main autode code, we always call allclose, so overriding __eq__ is only for coding simplicity? In that case it would be fine to use id pointer check.

Reply:
cheers @shoubhikraj. I tried the pointer equality but a bunch of tests failed so have gone with your idea 👍🏼